sparkle/src/components/Avatar.tsx

```diff
@@ -98,6 +99,16 @@ export function Avatar({
   disabled = false,
   className = "",
 }: AvatarProps) {
+  // Check if visual is a string and attempt to extract emoji and background color information from the url.
+  const emojiInfos =
+    typeof visual === "string" && getEmojiAndBackgroundFromUrl(visual);
+
+  const backgroundColorToUse = emojiInfos
```

Comment:
kind of a smell IMHO here (2 code paths, overlapping backgroundColors)
That + the fact that as you mentioned the URL will not be callable via API

Maybe we should directly go for the catchall endpoint? so we always have a working url. Do you think that'd be long to do?

Reply:
What do you mean by 👇 ?
> (2 code paths, overlapping backgroundColors)

We could go for the catch all endpoint, but it's probably out-of-scope of this PR and less important since I have strong doubt that users consuming our API care about the url for now 😅, WDYT?

Reply:
this was 2 separate things
- overlapping backgroundColors = the caller of avatar may specify a background color that may be ignored if the url is emoji. facilitates errors, the caller has to be aware of the implementation subtleties here
- 2 code paths: having to distinguish between emoji / not emoji in multiple places below (regular path / emoji path), although maybe path was not the right term
The fact that we now need to have a "visual" and a "visualToUse" also feels a bit weird IMHO

the code would be IMHO cleaner with only a url, that's why I was asking how long the catch all you had in mind would take---because if takes less than ~30mn it'd be worth doing right away IMHO

Reply:
> overlapping backgroundColors = the caller of avatar may specify a background color that may be ignored if the url is emoji. facilitates errors, the caller has to be aware of the implementation subtleties here

This was already the case, visual was taking precedence over any other things. I'm happy to improve the type to reflect this in the component's props, but this PR does not change anything on this point.

> the code would be IMHO cleaner with only a url, that's why I was asking how long the catch all you had in mind would take---because if takes less than ~30mn it'd be worth doing right away IMHO

> 2 code paths: having to distinguish between emoji / not emoji in multiple places below (regular path / emoji path), although maybe path was not the right term
The fact that we now need to have a "visual" and a "visualToUse" also feels a bit weird IMHO

The Avatar component has not been changed in a very long time, so we do centralize the logic of emoji or regular URL in one place 🤔.

For sure it's more than 30 minutes of work 🙃.